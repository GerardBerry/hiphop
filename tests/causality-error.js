"use hopscript"

var hh = require("hiphop");

var example = <hh.module>
  <hh.outputsignal name=I />
  <hh.outputsignal name=O />
  <hh.loop>
    <hh.sequence>
      <hh.present signal="O">
	<hh.emit signal="I"/>
      </hh.present>
      <hh.emit signal="O"/>
      <hh.pause/>
    </hh.sequence>
  </hh.loop>
</hh.module>;

exports.prg = new hh.ReactiveMachine(example, "presentemit");
