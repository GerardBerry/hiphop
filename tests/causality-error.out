presentemit> ;
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!! Module___I_pre_reg undefined
    Module___I undefined
!!! Module___I undefined
    Emit___I_go___0 undefined
!!! Module___O_pre_reg undefined
    Module___O undefined
!!! Module___O undefined
    Emit___O_go___0 undefined
!!! Emit___I_go___0 undefined
    If___and_then___0 undefined
!!! Nothing___go___0 undefined
    If___and_else___0 undefined
!!! If___and_then___0 undefined
    If___testexpr___0 undefined
!!! If___and_else___0 undefined
    If___testexpr___0 undefined
!!! If___testexpr___0 undefined
    Module___O undefined
!!! If___testexpr___1 undefined
    Module___O undefined
!!! If___k0_buffer___0 undefined
    Emit___I_go___0 undefined
!!! If___k0_buffer___0 undefined
    Nothing___go___0 undefined
!!! If___k undefined
    If___k0_buffer___0 undefined
    If___k0_buffer___0 undefined
!!! Emit___O_go___0 undefined
    If___k undefined
!!! Pause___reg undefined
    Pause___or_to_reg undefined
!!! Pause___or_to_reg undefined
    Pause___and___0 undefined
!!! Pause___go___0 undefined
    Emit___O_go___0 undefined
!!! Pause___and___0 undefined
    Pause___go___0 undefined
!!! Sequence___k1_buffer___0 undefined
    Pause___go___0 undefined
!!! Sequence___buf_k1_buffer_output___0 undefined
    Sequence___k1_buffer___0 undefined
!!! Loop___k1_buffer___0 undefined
    Sequence___buf_k1_buffer_output___0 undefined
!!! Module___global_k1 undefined
    Loop___k1_buffer___0 undefined
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
*** CAUSALITY ERROR
*** missing 19 nets
presentemit> ;
presentemit> ;
presentemit> ;
